I can't compile polybar

I tried different strategies to compile polybar, however, I get the following problem:
```
-- The CXX compiler identification is GNU 7.5.0
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No build type specified; using Release
-- Couldn't locate ccache, disabling ccache...
-- Performing Test HAS_SUGGEST_OVERRIDE
-- Performing Test HAS_SUGGEST_OVERRIDE - Success
-- Using supported compiler GNU-7.5.0
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Checking for module 'cairo-fc'
--   Found cairo-fc, version 1.15.10
-- Found CairoFC: /usr/include/cairo;/usr/include/glib-2.0;/usr/lib/x86_64-linux-gnu/glib-2.0/include;/usr/include/pixman-1;/usr/include/freetype2;/usr/include/libpng16;/usr/include/freetype2;/usr/include/libpng16 (found version "1.15.10") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target "Cairo::CairoFC".
Call Stack (most recent call first):
  cmake/modules/FindCairoFC.cmake:13 (create_imported_target)
  cmake/libpoly.cmake:49 (find_package)
  CMakeLists.txt:49 (include)


-- Found ALSA: /usr/lib/x86_64-linux-gnu/libasound.so (found version "1.1.3") 
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so (found version "7.58.0") 
-- Checking for module 'libmpdclient'
--   Found libmpdclient, version 2.11
-- Found LibMPDClient: /usr/include (found version "2.11") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target
  "LibMPDClient::LibMPDClient".
Call Stack (most recent call first):
  cmake/modules/FindLibMPDClient.cmake:13 (create_imported_target)
  cmake/libpoly.cmake:62 (find_package)
  CMakeLists.txt:49 (include)


-- Checking for module 'libnl-genl-3.0'
--   Found libnl-genl-3.0, version 3.2.29
-- Found LibNlGenl3: /usr/include/libnl3 (found version "3.2.29") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target
  "LibNlGenl3::LibNlGenl3".
Call Stack (most recent call first):
  cmake/modules/FindLibNlGenl3.cmake:13 (create_imported_target)
  cmake/libpoly.cmake:68 (find_package)
  CMakeLists.txt:49 (include)


-- Checking for module 'libpulse'
--   Found libpulse, version 11.1
-- Found LibPulse: /usr/include (found version "11.1") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target
  "LibPulse::LibPulse".
Call Stack (most recent call first):
  cmake/modules/FindLibPulse.cmake:13 (create_imported_target)
  cmake/libpoly.cmake:76 (find_package)
  CMakeLists.txt:49 (include)


-- Checking for module 'xcb-randr'
--   Found xcb-randr, version 1.13
-- Found Xcb_RANDR: /usr/include (found suitable version "1.13", minimum required is "1.12") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target "Xcb::RANDR".
Call Stack (most recent call first):
  cmake/modules/FindXcb.cmake:55 (create_imported_target)
  cmake/libpoly.cmake:100 (find_package)
  CMakeLists.txt:49 (include)


-- Checking for module 'xcb-composite'
--   Found xcb-composite, version 1.13
-- Found Xcb_COMPOSITE: /usr/include (found version "1.13") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target "Xcb::COMPOSITE".
Call Stack (most recent call first):
  cmake/modules/FindXcb.cmake:55 (create_imported_target)
  cmake/libpoly.cmake:101 (find_package)
  CMakeLists.txt:49 (include)


-- Checking for module 'xcb-xkb'
--   Found xcb-xkb, version 1.13
-- Found Xcb_XKB: /usr/include (found version "1.13") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target "Xcb::XKB".
Call Stack (most recent call first):
  cmake/modules/FindXcb.cmake:55 (create_imported_target)
  cmake/libpoly.cmake:101 (find_package)
  CMakeLists.txt:49 (include)


-- Checking for module 'xcb-cursor'
--   Found xcb-cursor, version 0.1.1
-- Found Xcb_CURSOR: /usr/include (found version "0.1.1") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target "Xcb::CURSOR".
Call Stack (most recent call first):
  cmake/modules/FindXcb.cmake:55 (create_imported_target)
  cmake/libpoly.cmake:101 (find_package)
  CMakeLists.txt:49 (include)


-- Checking for module 'xcb-xrm'
--   Found xcb-xrm, version 1.0
-- Found Xcb_XRM: /usr/include (found version "1.0") 
CMake Error at cmake/common/utils.cmake:121 (target_include_directories):
  Cannot specify include directories for imported target "Xcb::XRM".
Call Stack (most recent call first):
  cmake/modules/FindXcb.cmake:55 (create_imported_target)
  cmake/libpoly.cmake:101 (find_package)
  CMakeLists.txt:49 (include)


-- The C compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Checking for module 'xcb-proto'
--   Found xcb-proto, version 1.13
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.9", minimum required is "3.5") 
-- XCB[XCB]: Found component XCB
-- Found XCB_XCB: /usr/lib/x86_64-linux-gnu/libxcb.so  
-- XCB[ICCCM]: Found component ICCCM
-- Found XCB_ICCCM: /usr/lib/x86_64-linux-gnu/libxcb-icccm.so  
-- XCB[EWMH]: Found component EWMH
-- Found XCB_EWMH: /usr/lib/x86_64-linux-gnu/libxcb-ewmh.so  
-- XCB[UTIL]: Found component UTIL
-- Found XCB_UTIL: /usr/lib/x86_64-linux-gnu/libxcb-util.so  
-- XCB[IMAGE]: Found component IMAGE
-- Found XCB_IMAGE: /usr/lib/x86_64-linux-gnu/libxcb-image.so  
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so  
-- Searching for xcbgen with /usr/bin/python3
-- Searching for xcbgen with python3
-- Searching for xcbgen with python
-- Found xcbgen in /usr/lib/python2.7/dist-packages
-- XCB[RANDR]: Found component RANDR
-- Found XCB_RANDR: /usr/lib/x86_64-linux-gnu/libxcb-randr.so  
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so  
-- XCB[COMPOSITE]: Found component COMPOSITE
-- Found XCB_COMPOSITE: /usr/lib/x86_64-linux-gnu/libxcb-composite.so  
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so;/usr/lib/x86_64-linux-gnu/libxcb-composite.so  
-- XCB[XKB]: Found component XKB
-- Found XCB_XKB: /usr/lib/x86_64-linux-gnu/libxcb-xkb.so  
-- Found XCB: /usr/lib/x86_64-linux-gnu/libxcb.so;/usr/lib/x86_64-linux-gnu/libxcb-icccm.so;/usr/lib/x86_64-linux-gnu/libxcb-ewmh.so;/usr/lib/x86_64-linux-gnu/libxcb-util.so;/usr/lib/x86_64-linux-gnu/libxcb-image.so;/usr/lib/x86_64-linux-gnu/libxcb-randr.so;/usr/lib/x86_64-linux-gnu/libxcb-composite.so;/usr/lib/x86_64-linux-gnu/libxcb-xkb.so  
-- xpp: including xcb proto composite.xml
-- xpp: including xcb proto randr.xml
-- xpp: including xcb proto xkb.xml
-- xpp: including xcb proto xproto.xml
-- Checking for module 'jsoncpp'
--   Found jsoncpp, version 1.7.4
-- Font not found: fixed:pixelsize=10
-- Font not found: unifont:fontformat=truetype
-- Font not found: siji:pixelsize=10
--  Build:
--    Version: 3.5.4-103-gcd71b96c
--    Type: Release
--    CXX: /usr/bin/c++  -Wall -Wextra -Wpedantic -Wsuggest-override -O3 -DNDEBUG
--    LD: /usr/bin/ld  
--    sphinx-build: /usr/bin/sphinx-build 
--  Install Paths:
--    PREFIX:  /usr/local
--    BINDIR:  /usr/local/bin
--    DATADIR: /usr/local/share
--    DOCDIR:  /usr/local/share/doc/polybar
--    MANDIR:  /usr/local/share/man
--  Targets:
-- [X]   polybar
-- [X]   polybar-msg
-- [ ]   testsuite
-- [X]   documentation
-- [X]      html
-- [X]      man
-- [X]   sample config
-- [X]   shell files
--  Module support:
-- [X]   alsa (1.1.3)
-- [X]   curl (7.58.0)
-- [X]   i3
-- [X]   mpd (2.11)
-- [X]   network (libnl) (3.2.29)
-- [X]   pulseaudio (11.1)
-- [X]   xkeyboard (1.13)
--  X extensions:
-- [X]   xcb-randr (1.13)
-- [X]   xcb-randr (monitor support) (1.13)
-- [X]   xcb-composite (1.13)
-- [X]   xcb-xkb (1.13)
-- [X]   xcb-xrm (1.0)
-- [X]   xcb-cursor (0.1.1)
--  Log options:
-- [X]   Trace logging
-- Configuring incomplete, errors occurred!
See also "/home/sander/Downloads/polybar/polybar/build/CMakeFiles/CMakeOutput.log".
See also "/home/sander/Downloads/polybar/polybar/build/CMakeFiles/CMakeError.log".
 ```
I tried with the clang, clang++, g++, etc and I get the same problem.

I got the following with the PR you linked me:

Scanning dependencies of target doc_html
[  1%] Generating generated-sources/include/xpp/proto/composite.hpp
[  2%] Generating generated-sources/include/xpp/proto/xkb.hpp
Scanning dependencies of target polybar-msg
Scanning dependencies of target doc_man
[  3%] Generating generated-sources/include/xpp/proto/randr.hpp
[  4%] Generating generated-sources/include/xpp/proto/x.hpp
Scanning dependencies of target i3ipc++
[  4%] sphinx-build html: see doc/builder-html.log
[  5%] Building CXX object bin/CMakeFiles/polybar-msg.dir/ipc.cpp.o
[  6%] sphinx-build man: see doc/builder-man.log
[  7%] Building CXX object lib/i3ipcpp/CMakeFiles/i3ipc++.dir/src/ipc.cpp.o
[  8%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/env.cpp.o

Configuration error:
There is a syntax error in your configuration file: invalid syntax (conf.py, line 234)
list: size, ['red', 'green', 'blue']
doc/CMakeFiles/doc_html.dir/build.make:76: recipe for target 'doc/CMakeFiles/doc_html' failed
make[2]: *** [doc/CMakeFiles/doc_html] Error 1
CMakeFiles/Makefile2:477: recipe for target 'doc/CMakeFiles/doc_html.dir/all' failed
make[1]: *** [doc/CMakeFiles/doc_html.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

Configuration error:
There is a syntax error in your configuration file: invalid syntax (conf.py, line 234)
list: size, ['red', 'green', 'blue']
[  9%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/file.cpp.o
doc/CMakeFiles/doc_man.dir/build.make:76: recipe for target 'doc/CMakeFiles/doc_man' failed
make[2]: *** [doc/CMakeFiles/doc_man] Error 1
CMakeFiles/Makefile2:450: recipe for target 'doc/CMakeFiles/doc_man.dir/all' failed
make[1]: *** [doc/CMakeFiles/doc_man.dir/all] Error 2
[ 10%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/string.cpp.o
[ 11%] Building CXX object lib/i3ipcpp/CMakeFiles/i3ipc++.dir/src/ipc-util.cpp.o
[ 12%] Linking CXX executable polybar-msg
[ 12%] Built target polybar-msg
[ 13%] Linking CXX static library libi3ipc++.a
[ 13%] Built target i3ipc++
Scanning dependencies of target xpp
[ 14%] Linking CXX static library libxpp.a
[ 14%] Built target xpp
Makefile:148: recipe for target 'all' failed
make: *** [all] Error 2

I am going through a similar issue except after building with #2388 I get the following output.
```
Scanning dependencies of target doc_man
Scanning dependencies of target doc_html
[  1%] sphinx-build man: see doc/builder-man.log
[  1%] sphinx-build html: see doc/builder-html.log
[  1%] Built target doc_man
[  2%] Generating generated-sources/include/xpp/proto/x.hpp
[  2%] Built target doc_html
[  3%] Generating generated-sources/include/xpp/proto/composite.hpp
[  4%] Generating generated-sources/include/xpp/proto/randr.hpp
list: size, ['red', 'green', 'blue']
list: size, ['red', 'green', 'blue']
Scanning dependencies of target i3ipc++
[  5%] Building CXX object lib/i3ipcpp/CMakeFiles/i3ipc++.dir/src/ipc.cpp.o
[  6%] Building CXX object lib/i3ipcpp/CMakeFiles/i3ipc++.dir/src/ipc-util.cpp.o
[  7%] Generating generated-sources/include/xpp/proto/xkb.hpp
Scanning dependencies of target xpp
[  8%] Linking CXX static library libxpp.a
[  8%] Built target xpp
Scanning dependencies of target polybar-msg
[  9%] Building CXX object bin/CMakeFiles/polybar-msg.dir/ipc.cpp.o
[ 10%] Linking CXX static library libi3ipc++.a
[ 10%] Built target i3ipc++
[ 11%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/env.cpp.o
Scanning dependencies of target doc
[ 11%] Built target doc
Scanning dependencies of target poly
[ 12%] Building CXX object bin/CMakeFiles/poly.dir/__/generated-sources/settings.cpp.o
[ 13%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/file.cpp.o
[ 14%] Building CXX object bin/CMakeFiles/poly.dir/cairo/utils.cpp.o
[ 15%] Building CXX object bin/CMakeFiles/poly.dir/components/bar.cpp.o
[ 16%] Building CXX object bin/CMakeFiles/polybar-msg.dir/utils/string.cpp.o
[ 17%] Linking CXX executable polybar-msg
[ 17%] Built target polybar-msg
[ 18%] Building CXX object bin/CMakeFiles/poly.dir/components/builder.cpp.o
[ 19%] Building CXX object bin/CMakeFiles/poly.dir/components/command_line.cpp.o
[ 20%] Building CXX object bin/CMakeFiles/poly.dir/components/config.cpp.o
[ 21%] Building CXX object bin/CMakeFiles/poly.dir/components/config_parser.cpp.o
[ 22%] Building CXX object bin/CMakeFiles/poly.dir/components/controller.cpp.o
[ 23%] Building CXX object bin/CMakeFiles/poly.dir/components/ipc.cpp.o
[ 24%] Building CXX object bin/CMakeFiles/poly.dir/components/logger.cpp.o
[ 25%] Building CXX object bin/CMakeFiles/poly.dir/components/renderer.cpp.o
[ 26%] Building CXX object bin/CMakeFiles/poly.dir/components/screen.cpp.o
[ 27%] Building CXX object bin/CMakeFiles/poly.dir/components/taskqueue.cpp.o
[ 28%] Building CXX object bin/CMakeFiles/poly.dir/drawtypes/animation.cpp.o
[ 29%] Building CXX object bin/CMakeFiles/poly.dir/drawtypes/iconset.cpp.o
[ 30%] Building CXX object bin/CMakeFiles/poly.dir/drawtypes/label.cpp.o
[ 30%] Building CXX object bin/CMakeFiles/poly.dir/drawtypes/progressbar.cpp.o
[ 31%] Building CXX object bin/CMakeFiles/poly.dir/drawtypes/ramp.cpp.o
[ 32%] Building CXX object bin/CMakeFiles/poly.dir/events/signal_emitter.cpp.o
[ 33%] Building CXX object bin/CMakeFiles/poly.dir/events/signal_receiver.cpp.o
[ 34%] Building CXX object bin/CMakeFiles/poly.dir/modules/backlight.cpp.o
[ 35%] Building CXX object bin/CMakeFiles/poly.dir/modules/battery.cpp.o
[ 36%] Building CXX object bin/CMakeFiles/poly.dir/modules/bspwm.cpp.o
[ 37%] Building CXX object bin/CMakeFiles/poly.dir/modules/counter.cpp.o
[ 38%] Building CXX object bin/CMakeFiles/poly.dir/modules/cpu.cpp.o
[ 39%] Building CXX object bin/CMakeFiles/poly.dir/modules/date.cpp.o
[ 40%] Building CXX object bin/CMakeFiles/poly.dir/modules/fs.cpp.o
[ 41%] Building CXX object bin/CMakeFiles/poly.dir/modules/ipc.cpp.o
[ 42%] Building CXX object bin/CMakeFiles/poly.dir/modules/memory.cpp.o
[ 43%] Building CXX object bin/CMakeFiles/poly.dir/modules/menu.cpp.o
[ 44%] Building CXX object bin/CMakeFiles/poly.dir/modules/meta/base.cpp.o
[ 45%] Building CXX object bin/CMakeFiles/poly.dir/modules/script.cpp.o
[ 46%] Building CXX object bin/CMakeFiles/poly.dir/modules/systray.cpp.o
[ 47%] Building CXX object bin/CMakeFiles/poly.dir/modules/temperature.cpp.o
[ 48%] Building CXX object bin/CMakeFiles/poly.dir/modules/text.cpp.o
[ 49%] Building CXX object bin/CMakeFiles/poly.dir/modules/xbacklight.cpp.o
[ 50%] Building CXX object bin/CMakeFiles/poly.dir/modules/xwindow.cpp.o
[ 50%] Building CXX object bin/CMakeFiles/poly.dir/modules/xworkspaces.cpp.o
[ 51%] Building CXX object bin/CMakeFiles/poly.dir/tags/action_context.cpp.o
[ 52%] Building CXX object bin/CMakeFiles/poly.dir/tags/context.cpp.o
[ 53%] Building CXX object bin/CMakeFiles/poly.dir/tags/dispatch.cpp.o
[ 54%] Building CXX object bin/CMakeFiles/poly.dir/tags/parser.cpp.o
[ 55%] Building CXX object bin/CMakeFiles/poly.dir/utils/actions.cpp.o
[ 56%] Building CXX object bin/CMakeFiles/poly.dir/utils/bspwm.cpp.o
[ 57%] Building CXX object bin/CMakeFiles/poly.dir/utils/color.cpp.o
[ 58%] Building CXX object bin/CMakeFiles/poly.dir/utils/command.cpp.o
[ 59%] Building CXX object bin/CMakeFiles/poly.dir/utils/concurrency.cpp.o
[ 60%] Building CXX object bin/CMakeFiles/poly.dir/utils/env.cpp.o
[ 61%] Building CXX object bin/CMakeFiles/poly.dir/utils/factory.cpp.o
[ 62%] Building CXX object bin/CMakeFiles/poly.dir/utils/file.cpp.o
[ 63%] Building CXX object bin/CMakeFiles/poly.dir/utils/inotify.cpp.o
[ 64%] Building CXX object bin/CMakeFiles/poly.dir/utils/io.cpp.o
[ 65%] Building CXX object bin/CMakeFiles/poly.dir/utils/process.cpp.o
[ 66%] Building CXX object bin/CMakeFiles/poly.dir/utils/socket.cpp.o
[ 67%] Building CXX object bin/CMakeFiles/poly.dir/utils/string.cpp.o
[ 68%] Building CXX object bin/CMakeFiles/poly.dir/utils/throttle.cpp.o
[ 69%] Building CXX object bin/CMakeFiles/poly.dir/x11/atoms.cpp.o
[ 70%] Building CXX object bin/CMakeFiles/poly.dir/x11/background_manager.cpp.o
[ 70%] Building CXX object bin/CMakeFiles/poly.dir/x11/connection.cpp.o
[ 71%] Building CXX object bin/CMakeFiles/poly.dir/x11/ewmh.cpp.o
[ 72%] Building CXX object bin/CMakeFiles/poly.dir/x11/extensions/composite.cpp.o
[ 73%] Building CXX object bin/CMakeFiles/poly.dir/x11/extensions/randr.cpp.o
[ 74%] Building CXX object bin/CMakeFiles/poly.dir/x11/icccm.cpp.o
[ 75%] Building CXX object bin/CMakeFiles/poly.dir/x11/registry.cpp.o
[ 76%] Building CXX object bin/CMakeFiles/poly.dir/x11/tray_client.cpp.o
[ 77%] Building CXX object bin/CMakeFiles/poly.dir/x11/tray_manager.cpp.o
[ 78%] Building CXX object bin/CMakeFiles/poly.dir/x11/window.cpp.o
[ 79%] Building CXX object bin/CMakeFiles/poly.dir/x11/winspec.cpp.o
[ 80%] Building CXX object bin/CMakeFiles/poly.dir/x11/xembed.cpp.o
[ 81%] Building CXX object bin/CMakeFiles/poly.dir/adapters/alsa/control.cpp.o
[ 82%] Building CXX object bin/CMakeFiles/poly.dir/adapters/alsa/mixer.cpp.o
[ 83%] Building CXX object bin/CMakeFiles/poly.dir/modules/alsa.cpp.o
[ 84%] Building CXX object bin/CMakeFiles/poly.dir/modules/github.cpp.o
[ 85%] Building CXX object bin/CMakeFiles/poly.dir/utils/http.cpp.o
[ 86%] Building CXX object bin/CMakeFiles/poly.dir/modules/i3.cpp.o
[ 87%] Building CXX object bin/CMakeFiles/poly.dir/utils/i3.cpp.o
[ 88%] Building CXX object bin/CMakeFiles/poly.dir/adapters/mpd.cpp.o
[ 89%] Building CXX object bin/CMakeFiles/poly.dir/modules/mpd.cpp.o
[ 90%] Building CXX object bin/CMakeFiles/poly.dir/adapters/net.cpp.o
[ 90%] Building CXX object bin/CMakeFiles/poly.dir/modules/network.cpp.o
[ 91%] Building CXX object bin/CMakeFiles/poly.dir/adapters/net_nl.cpp.o
[ 92%] Building CXX object bin/CMakeFiles/poly.dir/adapters/pulseaudio.cpp.o
[ 93%] Building CXX object bin/CMakeFiles/poly.dir/modules/pulseaudio.cpp.o
[ 94%] Building CXX object bin/CMakeFiles/poly.dir/x11/cursor.cpp.o
[ 95%] Building CXX object bin/CMakeFiles/poly.dir/modules/xkeyboard.cpp.o
[ 96%] Building CXX object bin/CMakeFiles/poly.dir/x11/extensions/xkb.cpp.o
[ 97%] Building CXX object bin/CMakeFiles/poly.dir/x11/xresources.cpp.o
[ 98%] Linking CXX static library ../libs/libpoly.a
[ 98%] Built target poly
Scanning dependencies of target polybar
[ 99%] Building CXX object bin/CMakeFiles/polybar.dir/main.cpp.o
[100%] Linking CXX executable polybar
../libs/libpoly.a(alsa.cpp.o): In function `polybar::modules::alsa_module::teardown()':
alsa.cpp:(.text+0x858): undefined reference to `snd_config_update_free_global'
../libs/libpoly.a(alsa.cpp.o): In function `polybar::modules::module<polybar::modules::alsa_module>::stop()':
alsa.cpp:(.text._ZN7polybar7modules6moduleINS0_11alsa_moduleEE4stopEv[_ZN7polybar7modules6moduleINS0_11alsa_moduleEE4stopEv]+0x229): undefined reference to `snd_config_update_free_global'
../libs/libpoly.a(http.cpp.o): In function `polybar::http_downloader::http_downloader(int)':
http.cpp:(.text+0x192): undefined reference to `curl_easy_init'
http.cpp:(.text+0x1ab): undefined reference to `curl_easy_setopt'
http.cpp:(.text+0x1c1): undefined reference to `curl_easy_setopt'
http.cpp:(.text+0x1d9): undefined reference to `curl_easy_setopt'
http.cpp:(.text+0x1ed): undefined reference to `curl_easy_setopt'
http.cpp:(.text+0x2ab): undefined reference to `curl_easy_setopt'
../libs/libpoly.a(http.cpp.o):http.cpp:(.text+0x2e7): more undefined references to `curl_easy_setopt' follow
../libs/libpoly.a(http.cpp.o): In function `polybar::http_downloader::get(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
http.cpp:(.text+0x5ec): undefined reference to `curl_easy_perform'
http.cpp:(.text+0x8bb): undefined reference to `curl_easy_setopt'
http.cpp:(.text+0xa6e): undefined reference to `curl_easy_strerror'
../libs/libpoly.a(http.cpp.o): In function `polybar::http_downloader::response_code()':
http.cpp:(.text+0xcb8): undefined reference to `curl_easy_getinfo'
../libs/libpoly.a(http.cpp.o): In function `polybar::http_downloader::~http_downloader()':
http.cpp:(.text+0x3d4): undefined reference to `curl_easy_cleanup'
../libs/libpoly.a(control.cpp.o): In function `polybar::alsa::control::wait(int)':
control.cpp:(.text+0x112): undefined reference to `snd_ctl_wait'
control.cpp:(.text+0x11c): undefined reference to `snd_ctl_event_sizeof'
control.cpp:(.text+0x131): undefined reference to `snd_ctl_event_sizeof'
control.cpp:(.text+0x14f): undefined reference to `snd_ctl_read'
control.cpp:(.text+0x15c): undefined reference to `snd_ctl_event_get_type'
control.cpp:(.text+0x184): undefined reference to `snd_ctl_event_elem_get_mask'
../libs/libpoly.a(control.cpp.o): In function `polybar::alsa::control::test_device_plugged()':
control.cpp:(.text+0x20e): undefined reference to `snd_ctl_elem_value_sizeof'
control.cpp:(.text+0x223): undefined reference to `snd_ctl_elem_value_sizeof'
control.cpp:(.text+0x241): undefined reference to `snd_hctl_elem_read'
control.cpp:(.text+0x250): undefined reference to `snd_ctl_elem_value_get_boolean'
../libs/libpoly.a(control.cpp.o): In function `polybar::alsa::control::control(int)':
control.cpp:(.text+0x30c): undefined reference to `snd_ctl_open'
control.cpp:(.text+0x31a): undefined reference to `snd_config_update_free_global'
control.cpp:(.text+0x327): undefined reference to `snd_hctl_open_ctl'
control.cpp:(.text+0x338): undefined reference to `snd_config_update_free_global'
control.cpp:(.text+0x341): undefined reference to `snd_hctl_load'
control.cpp:(.text+0x34f): undefined reference to `snd_ctl_elem_id_sizeof'
control.cpp:(.text+0x364): undefined reference to `snd_ctl_elem_id_sizeof'
control.cpp:(.text+0x37f): undefined reference to `snd_ctl_elem_id_set_numid'
control.cpp:(.text+0x384): undefined reference to `snd_ctl_elem_info_sizeof'
control.cpp:(.text+0x399): undefined reference to `snd_ctl_elem_info_sizeof'
control.cpp:(.text+0x3b5): undefined reference to `snd_ctl_elem_info_set_id'
control.cpp:(.text+0x3c1): undefined reference to `snd_ctl_elem_info'
control.cpp:(.text+0x3d5): undefined reference to `snd_ctl_elem_info_get_id'
control.cpp:(.text+0x3e1): undefined reference to `snd_hctl_find_elem'
control.cpp:(.text+0x3fc): undefined reference to `snd_ctl_subscribe_events'
control.cpp:(.text+0x52a): undefined reference to `snd_ctl_elem_id_get_numid'
control.cpp:(.text+0x5d1): undefined reference to `snd_ctl_elem_id_get_numid'
control.cpp:(.text+0x699): undefined reference to `snd_ctl_close'
../libs/libpoly.a(control.cpp.o): In function `polybar::alsa::control::~control()':
control.cpp:(.text+0xca): undefined reference to `snd_hctl_close'
../libs/libpoly.a(control.cpp.o): In function `void polybar::alsa::throw_exception<polybar::alsa::control_error>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, int)':
control.cpp:(.text._ZN7polybar4alsa15throw_exceptionINS0_13control_errorEEEvONSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEi[_ZN7polybar4alsa15throw_exceptionINS0_13control_errorEEEvONSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEi]+0x1e): undefined reference to `snd_strerror'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::~mixer()':
mixer.cpp:(.text+0xcd): undefined reference to `snd_mixer_close'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::get_volume()':
mixer.cpp:(.text+0x169): undefined reference to `snd_mixer_selem_get_playback_volume_range'
mixer.cpp:(.text+0x17f): undefined reference to `snd_mixer_selem_has_playback_channel'
mixer.cpp:(.text+0x198): undefined reference to `snd_mixer_selem_get_playback_volume'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::get_normalized_volume()':
mixer.cpp:(.text+0x25a): undefined reference to `snd_mixer_selem_get_playback_dB_range'
mixer.cpp:(.text+0x270): undefined reference to `snd_mixer_selem_has_playback_channel'
mixer.cpp:(.text+0x28a): undefined reference to `snd_mixer_selem_get_playback_dB'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::set_volume(float)':
mixer.cpp:(.text+0x40f): undefined reference to `snd_mixer_selem_has_playback_channel'
mixer.cpp:(.text+0x429): undefined reference to `snd_mixer_selem_get_playback_switch'
mixer.cpp:(.text+0x47f): undefined reference to `snd_mixer_selem_get_playback_volume_range'
mixer.cpp:(.text+0x4cc): undefined reference to `snd_mixer_selem_set_playback_volume_all'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::set_normalized_volume(float)':
mixer.cpp:(.text+0x51f): undefined reference to `snd_mixer_selem_has_playback_channel'
mixer.cpp:(.text+0x539): undefined reference to `snd_mixer_selem_get_playback_switch'
mixer.cpp:(.text+0x5ab): undefined reference to `snd_mixer_selem_get_playback_dB_range'
mixer.cpp:(.text+0x64c): undefined reference to `snd_mixer_selem_set_playback_dB_all'
mixer.cpp:(.text+0x67c): undefined reference to `snd_mixer_selem_set_playback_dB_all'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::toggle_mute()':
mixer.cpp:(.text+0x6c4): undefined reference to `snd_mixer_selem_get_playback_switch'
mixer.cpp:(.text+0x6d9): undefined reference to `snd_mixer_selem_set_playback_switch_all'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::is_muted()':
mixer.cpp:(.text+0x73f): undefined reference to `snd_mixer_selem_has_playback_channel'
mixer.cpp:(.text+0x759): undefined reference to `snd_mixer_selem_get_playback_switch'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::process_events()':
mixer.cpp:(.text+0x7ca): undefined reference to `snd_mixer_handle_events'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::mixer(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&)':
mixer.cpp:(.text+0x901): undefined reference to `snd_mixer_open'
mixer.cpp:(.text+0x90f): undefined reference to `snd_config_update_free_global'
mixer.cpp:(.text+0x91b): undefined reference to `snd_mixer_attach'
mixer.cpp:(.text+0x930): undefined reference to `snd_mixer_selem_register'
mixer.cpp:(.text+0x941): undefined reference to `snd_mixer_load'
mixer.cpp:(.text+0x94f): undefined reference to `snd_mixer_selem_id_sizeof'
mixer.cpp:(.text+0x968): undefined reference to `snd_mixer_selem_id_sizeof'
mixer.cpp:(.text+0x97f): undefined reference to `snd_mixer_selem_id_set_index'
mixer.cpp:(.text+0x98b): undefined reference to `snd_mixer_selem_id_set_name'
mixer.cpp:(.text+0x996): undefined reference to `snd_mixer_find_selem'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::wait(int)':
mixer.cpp:(.text+0xbbd): undefined reference to `snd_mixer_wait'
mixer.cpp:(.text+0xbca): undefined reference to `snd_mixer_handle_events'
../libs/libpoly.a(mixer.cpp.o): In function `polybar::alsa::mixer::set_mute(bool)':
mixer.cpp:(.text+0x699): undefined reference to `snd_mixer_selem_set_playback_switch_all'
../libs/libpoly.a(mixer.cpp.o): In function `void polybar::alsa::throw_exception<polybar::alsa::mixer_error>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, int)':
mixer.cpp:(.text._ZN7polybar4alsa15throw_exceptionINS0_11mixer_errorEEEvONSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEi[_ZN7polybar4alsa15throw_exceptionINS0_11mixer_errorEEEvONSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEi]+0x1e): undefined reference to `snd_strerror'
collect2: error: ld returned 1 exit status
bin/CMakeFiles/polybar.dir/build.make:105: recipe for target 'bin/polybar' failed
make[2]: *** [bin/polybar] Error 1
CMakeFiles/Makefile2:589: recipe for target 'bin/CMakeFiles/polybar.dir/all' failed
make[1]: *** [bin/CMakeFiles/polybar.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/cmake/common/utils.cmake
+++ b/cmake/common/utils.cmake
@@ -117,8 +117,8 @@ function(create_imported_target library_name includes libraries)
   add_library(${library_name} INTERFACE IMPORTED)
   set_target_properties(${library_name} PROPERTIES
     INTERFACE_LINK_LIBRARIES "${libraries}"
+    INTERFACE_INCLUDE_DIRECTORIES "${includes}"
   )
-  target_include_directories(${library_name} SYSTEM INTERFACE ${includes})
 endfunction()
 # }}}
 # checklib {{{

--- a/cmake/libpoly.cmake
+++ b/cmake/libpoly.cmake
@@ -50,12 +50,10 @@ find_package(CairoFC REQUIRED)
 
 if (ENABLE_ALSA)
   find_package(ALSA REQUIRED)
-  set(ALSA_VERSION ${ALSA_VERSION_STRING})
 endif()
 
 if (ENABLE_CURL)
   find_package(CURL REQUIRED)
-  set(CURL_VERSION ${CURL_VERSION_STRING})
 endif()
 
 if (ENABLE_MPD)

--- a/cmake/modules/FindALSA.cmake
+++ b/cmake/modules/FindALSA.cmake
@@ -1,0 +1,14 @@
+# This module defines an imported target `ALSA::ALSA` if alsa is found
+#
+# Defines the following Variables (see find_package_impl for more info):
+# ALSA_FOUND
+# ALSA_INCLUDE_DIR
+# ALSA_INCLUDE_DIRS
+# ALSA_LIBRARY
+# ALSA_LIBRARIES
+# ALSA_VERSION
+find_package_impl("alsa" "ALSA" "alsa/asoundlib.h")
+
+if(ALSA_FOUND AND NOT TARGET ALSA::ALSA)
+  create_imported_target("ALSA::ALSA" "${ALSA_INCLUDE_DIR}" "${ALSA_LIBRARY}")
+endif()

--- a/cmake/modules/FindCURL.cmake
+++ b/cmake/modules/FindCURL.cmake
@@ -1,0 +1,14 @@
+# This module defines an imported target `CURL::libcurl` if libcurl is found
+#
+# Defines the following Variables (see find_package_impl for more info):
+# CURL_FOUND
+# CURL_INCLUDE_DIR
+# CURL_INCLUDE_DIRS
+# CURL_LIBRARY
+# CURL_LIBRARIES
+# CURL_VERSION
+find_package_impl("libcurl" "CURL" "curl/curl.h")
+
+if(CURL_FOUND AND NOT TARGET CURL::libcurl)
+  create_imported_target("CURL::libcurl" "${CURL_INCLUDE_DIR}" "${CURL_LIBRARY}")
+endif()

--- a/include/utils/color.hpp
+++ b/include/utils/color.hpp
@@ -22,7 +22,7 @@ class rgba {
   bool operator==(const rgba& other) const;
 
   uint32_t value() const;
-  type type() const;
+  type get_type() const;
 
   double alpha_d() const;
   double red_d() const;

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -151,20 +151,59 @@ if(BUILD_LIBPOLY)
     Cairo::CairoFC
     moodycamel
     xpp
-    $<$<TARGET_EXISTS:i3ipc++>:i3ipc++>
-    $<$<TARGET_EXISTS:ALSA::ALSA>:ALSA::ALSA>
-    $<$<TARGET_EXISTS:CURL::libcurl>:CURL::libcurl>
-    $<$<TARGET_EXISTS:LibMPDClient::LibMPDClient>:LibMPDClient::LibMPDClient>
-    $<$<TARGET_EXISTS:LibNlGenl3::LibNlGenl3>:LibNlGenl3::LibNlGenl3>
-    $<$<TARGET_EXISTS:Libiw::Libiw>:Libiw::Libiw>
-    $<$<TARGET_EXISTS:LibPulse::LibPulse>:LibPulse::LibPulse>
-    $<$<TARGET_EXISTS:Xcb::RANDR>:Xcb::RANDR>
-    $<$<TARGET_EXISTS:Xcb::COMPOSITE>:Xcb::COMPOSITE>
-    $<$<TARGET_EXISTS:Xcb::XKB>:Xcb::XKB>
-    $<$<TARGET_EXISTS:Xcb::CURSOR>:Xcb::CURSOR>
-    $<$<TARGET_EXISTS:Xcb::XRM>:Xcb::XRM>
-    $<$<TARGET_EXISTS:LibInotify::LibInotify>:LibInotify::LibInotify>
     )
+
+  if (TARGET i3ipc++)
+    target_link_libraries(poly PUBLIC i3ipc++)
+  endif()
+
+  if (TARGET ALSA::ALSA)
+    target_link_libraries(poly PUBLIC ALSA::ALSA)
+  endif()
+
+  if (TARGET CURL::libcurl)
+    target_link_libraries(poly PUBLIC CURL::libcurl)
+  endif()
+
+  if (TARGET LibMPDClient::LibMPDClient)
+    target_link_libraries(poly PUBLIC LibMPDClient::LibMPDClient)
+  endif()
+
+  if (TARGET LibNlGenl3::LibNlGenl3)
+    target_link_libraries(poly PUBLIC LibNlGenl3::LibNlGenl3)
+  endif()
+
+  if (TARGET Libiw::Libiw)
+    target_link_libraries(poly PUBLIC Libiw::Libiw)
+  endif()
+
+  if (TARGET LibPulse::LibPulse)
+    target_link_libraries(poly PUBLIC LibPulse::LibPulse)
+  endif()
+
+  if (TARGET Xcb::RANDR)
+    target_link_libraries(poly PUBLIC Xcb::RANDR)
+  endif()
+
+  if (TARGET Xcb::COMPOSITE)
+    target_link_libraries(poly PUBLIC Xcb::COMPOSITE)
+  endif()
+
+  if (TARGET Xcb::XKB)
+    target_link_libraries(poly PUBLIC Xcb::XKB)
+  endif()
+
+  if (TARGET Xcb::CURSOR)
+    target_link_libraries(poly PUBLIC Xcb::CURSOR)
+  endif()
+
+  if (TARGET Xcb::XRM)
+    target_link_libraries(poly PUBLIC Xcb::XRM)
+  endif()
+
+  if (TARGET LibInotify::LibInotify)
+    target_link_libraries(poly PUBLIC LibInotify::LibInotify)
+  endif()
 
   target_compile_options(poly PUBLIC $<$<CXX_COMPILER_ID:GNU>:$<$<CONFIG:MinSizeRel>:-flto>>)
   set_target_properties(poly PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/libs)

--- a/src/utils/color.cpp
+++ b/src/utils/color.cpp
@@ -106,7 +106,7 @@ uint32_t rgba::value() const {
   return this->m_value;
 }
 
-enum rgba::type rgba::type() const {
+enum rgba::type rgba::get_type() const {
   return m_type;
 }
 

--- a/tests/unit_tests/utils/color.cpp
+++ b/tests/unit_tests/utils/color.cpp
@@ -12,7 +12,7 @@ TEST(Rgba, constructor) {
   EXPECT_FALSE(rgba("#-abc").has_color());
   EXPECT_FALSE(rgba("#xyz").has_color());
 
-  EXPECT_EQ(rgba::type::ALPHA_ONLY, rgba{"#12"}.type());
+  EXPECT_EQ(rgba::type::ALPHA_ONLY, rgba{"#12"}.get_type());
 
   EXPECT_EQ(0xff000000, rgba{"#ff"}.value());
 

